galley/pkg/runtime/monitoring/monitoring.go

```diff
@@ -124,6 +151,36 @@ func RecordStateTypeCountWithContext(ctx context.Context, count int) {
 	}
 }
 
+// RecordDetailedStateType
+func RecordDetailedStateType(namespace, name string, collection resource.Collection, count int) {
+	ctx, err := tag.New(context.Background(), tag.Insert(NamespaceTag, namespace),
+		tag.Insert(NameTag, name))
+	if err != nil {
+		log.Scope.Errorf("Error creating monitoring context for counting state: %v", err)
+	} else {
+		RecordDetailedStateTypeWithContext(ctx, collection, count)
+	}
+}
+
+// RecordDetailedStateTypeWithContext
+func RecordDetailedStateTypeWithContext(ctx context.Context, collection resource.Collection, count int) {
+	if ctx == nil {
+		return
```

Comment:
Should we default the tags we need, in cases when `ctx` is null?

Reply:
Namespace we can default to 'default' but what about name?